presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationCompiler.java

```diff
@@ -88,37 +94,35 @@ public static BindableAggregationFunction generateAggregationBindableFunction(Cl
         AggregationFunction aggregationAnnotation = aggregationDefinition.getAnnotation(AggregationFunction.class);
         requireNonNull(aggregationAnnotation, "aggregationAnnotation is null");
 
-        DynamicClassLoader classLoader = new DynamicClassLoader(aggregationDefinition.getClassLoader());
-
         ImmutableList.Builder<BindableAggregationFunction> builder = ImmutableList.builder();
 
         for (Class<?> stateClass : getStateClasses(aggregationDefinition)) {
-            AccumulatorStateSerializer<?> stateSerializer = new StateCompiler().generateStateSerializer(stateClass, classLoader);
-
             for (Method outputFunction : getOutputFunctions(aggregationDefinition, stateClass)) {
                 for (Method inputFunction : getInputFunctions(aggregationDefinition, stateClass)) {
                     List<LongVariableConstraint> longVariableConstraints = parseLongVariableConstraints(inputFunction);
+                    List<TypeVariableConstraint> typeVariableConstraints = parseTypeVariableConstraints(inputFunction);
 
                     for (String name : getNames(outputFunction, aggregationAnnotation)) {
                         List<TypeSignature> inputTypes = getInputTypesSignatures(inputFunction);
                         TypeSignature outputType = TypeSignature.parseTypeSignature(outputFunction.getAnnotation(OutputFunction.class).value());
-
+                        Signature signature = new Signature(
+                                name,
+                                aggregationAnnotation.approximate() ? FunctionKind.APPROXIMATE_AGGREGATE : FunctionKind.AGGREGATE,
```

Comment:
I've added ensuring that each commit builds to todo list. I will fix that in this step.